contexts/SecurityProvider.tsx

```diff
@@ -0,0 +1,305 @@
+            'AuthenticationType',
+            SecurityAuthenticationTypes.LOCALAUTH,
+          );
+          setSupportedType(SecurityAuthenticationTypes.LOCALAUTH);
+          setError(undefined);
+        } else {
+          AsyncStorage.setItem(
+            'AuthenticationType',
+            SecurityAuthenticationTypes.MANUAL,
+          );
+          setSupportedType(SecurityAuthenticationTypes.MANUAL);
+          setError(undefined);
+        }
+      } else {
+        if (
+          supportedBiometry === null &&
+          val == SecurityAuthenticationTypes.KEYCHAIN
+        ) {
+          // Keychain not available anymore
+          setError(
+            'Your previous authentication method is not available anymore. Please enroll your biometrics on your device again or reinstall the app.',
+          );
+        } else if (
+          isPinOrFingerprintSet !== true &&
+          val == SecurityAuthenticationTypes.LOCALAUTH
+        ) {
+          // LocalAuth not available anymore
+          setError(
+            'Your previous authentication method is not available anymore. Please add a PIN code lock to your device or reinstall the app.',
+          );
+        } else {
+          setSupportedType(val as SecurityAuthenticationTypes);
+          setError(undefined);
+        }
+      }
+    });
+  }, [isPinOrFingerprintSet, supportedBiometry]);
+
+  const Encrypt = (plain: string, key: string): string => {
+    const iv = Buffer.from(DeviceInfo.getUniqueId(), 'utf8').slice(0, 16);
+    const aes = crypto.createCipheriv(
+      'aes-256-cbc',
+      crypto.createHmac('sha256', key).digest(),
+      iv,
+    );
+    let ciphertext = aes.update(plain);
+    ciphertext = Buffer.concat([iv, ciphertext, aes.final()]);
+    return ciphertext.toString('base64');
+  };
+
+  const Decrypt = (cypher: string, key: string): string => {
+    const ciphertextBytes = Buffer.from(cypher, 'base64');
+    const iv = ciphertextBytes.slice(0, 16);
+    const data = ciphertextBytes.slice(16);
+    const aes = crypto.createDecipheriv(
+      'aes-256-cbc',
+      crypto.createHmac('sha256', key).digest(),
+      iv,
+    );
+    let plaintextBytes = Buffer.from(aes.update(data));
+    plaintextBytes = Buffer.concat([plaintextBytes, aes.final()]);
+    return plaintextBytes.toString();
+  };
+
+  const writeEncrypedStorage = useCallback(async (suffix: string): void => {
+    try {
+      await EncryptedStorage.setItem(
+        suffix,
+        crypto.randomBytes(64).toString('hex'),
+      );
+    } catch (e) {
+      console.log(e);
+    }
+  }, []);
+
+  const readEncrytedStorage = async (suffix: string): Promise<string> => {
+    try {
+      let creds = await EncryptedStorage.getItem(suffix);
+
+      if (creds) {
+        await new Promise((res, rej) => {
+          LocalAuth((error: any) => {
+            if (!error) {
+              res(true);
+            } else {
+              rej('failed local auth');
+            }
+          });
+        });
```

Comment:
can we use `const res = await LocalAuth();` directly?

Reply:
from the looks of it, I'd say LocalAuth does not return a promise but takes a callback as argument